lib/bundler/dsl.rb

```diff
@@ -332,7 +332,20 @@ def normalize_options(name, version, opts)
       install_if = @install_conditionals.dup
       install_if.concat Array(opts.delete("install_if"))
       install_if = install_if.reduce(true) do |memo, val|
-        memo && (val.respond_to?(:call) ? val.call : val)
+        memo && case val
+                when TrueClass, FalseClass, NilClass
+                  val
+                when Proc
+                  val.call
+                when String
+                  begin
+                    eval(val, TOPLEVEL_BINDING) # rubocop:disable Lint/Eval
+                  rescue
+                    true
+                  end
+                else
+                  true
+                end
```

Comment:
I can move this commit to a separate PR, sure. I think this was actually a remnant of me experimenting with something else